// uu5g04/dist-node/bricks/modal-body.js
import React from 'react';
import {BaseMixin, ElementaryMixin, ContentMixin} from '../common/common.js';

import './modal-body.less';

export default React.createClass({

  //@@viewOn:mixins
  mixins: [
    BaseMixin,
    ElementaryMixin,
    ContentMixin
  ],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'UU5.Bricks.Modal.Body',
    classNames: {
      main: 'uu5-bricks-modal-body modal-body'
    },
    defaults: {
      parentTagName: 'UU5.Bricks.Modal'
    },
    errors: {
      invalidParent: 'Parent of this component is not Modal.'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  componentWillMount: function () {
    this.checkParentTagName(this.getDefault().parentTagName);

    let parent = this.getParentByType('isModal');
    if (!parent) {
      this.showError("invalidParent");
    }
  },
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  shouldComponentUpdate(nextProps, nextState) {
    return this.shouldRender(nextProps, nextState);
  },
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function () {
    return (
      <div {...this.buildMainAttrs()}>
        {this.getChildren()}
        {this.getDisabledCover()}
      </div>
    );
  }
  //@@viewOff:render
});
