// uu5g04/dist-node/bricks/progress-bar-cover.js
import React from 'react';
import {BaseMixin, ElementaryMixin} from '../common/common.js';

import './progress-bar-cover.less';

export default React.createClass({

  //@@viewOn:mixins
  mixins: [
    BaseMixin,
    ElementaryMixin
  ],
  //@@viewOff:mixins

  //@@viewOn:statics
  statics: {
    tagName: 'UU5.Bricks.ProgressBar.Cover',
    classNames: {
      main: 'uu5-bricks-progress-bar-cover progress'
    },
    errors: {
      invalidParent: 'Parent of this component is not ProgressBar.'
    }
  },
  //@@viewOff:statics

  //@@viewOn:propTypes
  //@@viewOff:propTypes

  //@@viewOn:getDefaultProps
  //@@viewOff:getDefaultProps

  //@@viewOn:standardComponentLifeCycle
  componentWillMount: function () {
    if (!(this.getParent() && this.getParent().isProgressBar)) {
      this.showError("invalidParent");
    }
  },

  shouldComponentUpdate(nextProps, nextState) {
    return this.shouldRender(nextProps, nextState);
  },
  //@@viewOff:standardComponentLifeCycle

  //@@viewOn:interface
  //@@viewOff:interface

  //@@viewOn:overridingMethods
  //@@viewOff:overridingMethods

  //@@viewOn:componentSpecificHelpers
  //@@viewOff:componentSpecificHelpers

  //@@viewOn:render
  render: function () {
    return (
      <div {...this.buildMainAttrs()}>
        {this.props.children && React.Children.toArray(this.props.children)}
        {this.getDisabledCover()}
      </div>
    );
  }
  //@@viewOff:render
});
